### PR TITLE
[tests] Rework questions/options tests

### PR DESCRIPTION
**Pull request by Axolotle**
Rework the questions/options tests and add multiple missing option types tests scenarios.

There's some tests that currently passes but probably shouldn't, some that do not but probably should.
You can search for `xfail`, `xpass`, and some `FIXME`s to have a quick look at those.
Plan is to fix the unexpected results on the pydantic migration if you agree with my comments.
There's a bunch of problems with nones () where things are not very clear what final None value we want.

Also added some quick fixes in `utils/config.py` to ease tests. @zamentur do you see any potential problems with those modifications? There might be magic stuff in bash helpers that I'm not aware of.

## PR Status

ok?

## How to test

```bash
pytest src/tests/test_questions.py
```

### CODE REVIEW

File: src/utils/config.py [hunk 1]
```diff
@@ -1481,8 +1505,12 @@ def _prevalidate(self):
 
         super()._prevalidate()
 
+        # Validation should have already failed if required
+        if self.value in (None, ""):
```
Comment (alexAubin) [inline src/utils/config.py:1509]:
```suggestion
        if self.value in [None, ""]:
```

File: src/utils/config.py [hunk 2]
```diff
@@ -1481,8 +1505,12 @@ def _prevalidate(self):
...
+            return self.value
+
         if Moulinette.interface.type != "api":
-            if not self.value or not os.path.exists(str(self.value)):
+            if not os.path.exists(str(self.value)) or not os.path.isfile(str(self.value)):
```
Comment (alexAubin) [inline src/utils/config.py:1513]: hmm what's the motivation for this change here ? Not a huge deal but why not keep the `if not self.value` ? And Why enforce `isfile` ? (`exists` should be enough ... I guess people *could* maybe want to use this kind of arg for a directory or a symlink ... ?). Though as said, not a huge deal but I'm confused :sweat_smile:

Comment (Axolotle) [reply src/utils/config.py:1513]: Hum, I removed the `if not self.value` because it was raising the error even if it was optional in cli, hence the check for it before. For the enforcing I don't remember lel, i'll run the tests without to check
